tests/test_phase0.py

```
@@ -371,6 +372,65 @@ def test_memory_dropout_not_in_inference(self):
+        mask = create_sliding_window_mask(seq_len=8, window_size=8, device=torch.device("cpu"))
+        # With window=seq_len, it's just a causal mask (no sliding window constraint)
+        expected_causal = torch.triu(torch.ones(8, 8, dtype=torch.bool), diagonal=1)
+        assert torch.equal(mask, expected_causal)
+
+    def test_sliding_window_mask_window_constraint(self):
+        """Positions beyond window should be masked."""
+        from src.model.skeleton import create_sliding_window_mask
+
+        mask = create_sliding_window_mask(seq_len=6, window_size=3, device=torch.device("cpu"))
+        # Position 3 should NOT see position 0 (0 < 3 - 3 + 1 = 1)
+        assert mask[3, 0] == True  # masked
```

Ruff (0.14.14)

Avoid equality comparisons to `True`; use `mask[3, 0]:` for truth checks

Replace with `mask[3, 0]`

(E712)